aten/src/ATen/core/Generator.h

```diff
@@ -1,24 +1,62 @@
 #pragma once
 
-#include <ATen/core/ATenGeneral.h>
 #include <stdint.h>
+#include <memory>
+#include <mutex>
+#include <deque>
+#include <atomic>
+#include <typeinfo>
+#include <utility>
+
+#include <c10/util/Exception.h>
+#include <c10/util/C++17.h>
+#include <ATen/Device.h>
+
+/*
+* Generator note.
+* A Pseudo Random Number Generator (PRNG) is an engine that uses an algorithm to 
+* generate a seemingly random sequence of numbers, that may be later be used in creating 
+* a random distribution. Such an engine almost always maintains a state and requires a
+* seed to start off the creation of random numbers. Often times, users have
+* encountered that it could be beneficial to be able to create, retain, and destroy 
+* PRNG states and also be able to have control over the seed value.
+*
+* A Generator in ATen gives users the ability to read, write and modify a PRNG engine.
+* For instance, it does so by letting users seed a PRNG engine, fork the state of the
+* engine, etc.
+*
+* By default, there is one generator per device, and a device's generator is 
+* lazily created. A user can use the torch.Generator() api to create their own generator.
+* Currently torch.Generator() can only create a CPUGenerator.
+*/
 
 namespace at {
 
+constexpr uint64_t default_rng_seed_val = 67280421310721;
```

Comment:
OOC, where did we get this number from?

Reply:
So since the Philox engine has two 32-bit keys (that is it's state is 64 bit I think?), it is recommended that a RNG engine should be seeded with the same number of bits as its state (I think I read this somewhere in one of the links I posted in the last PR). Also the seed value should have a good distribution of 0s and 1s in the bit representation. So I chose a large prime number that can be fit into uint64_t: https://primes.utm.edu/curios/page.php/67280421310721.html. You'll notice another number in:
https://github.com/pytorch/pytorch/blob/2b826c6171608bebd96be6dd1f8201ab6a0a8dff/test/common_utils.py#L46-L48
python side won't let me seed with 64 bit? So I just chose a large prime there under 32 bit.  
 ¯_(ツ)_/¯